tests/integration/targets/aws_ses_identity_dkim/tasks/main.yml

```diff
@@ -0,0 +1,160 @@
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert successful result
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_attributes.dkim_enabled == True
+          - email_result.dkim_attributes.dkim_tokens|length > 0
+
+    ## DKIM enable idempotence test (rerun)
+
+    - name: Enable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert no change (itempotence)
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_attributes.dkim_enabled == True
+
+    - name: Enable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: enabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert no change (idempotence)
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_attributes.dkim_enabled == True
+
+    ## DKIM disable test
+
+    - name: Disable DKIM verification on domain identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - domain_result.changed == True
+          - domain_result.dkim_attributes.dkim_enabled == False
+
+    - name: Disable DKIM verification on email identity
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - email_result.changed == True
+          - email_result.dkim_attributes.dkim_enabled == False
+
+    ## DKIM disable idempotence test (rerun)
+
+    - name: Disable DKIM verification on domain identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_domain }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: domain_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - domain_result.changed == False
+          - domain_result.dkim_attributes.dkim_enabled == False
+
+    - name: Disable DKIM verification on email identity (rerun)
+      aws_ses_identity_dkim:
+        identity: "{{ dkim_identity_email }}"
+        state: disabled
+        <<: *aws_connection_info
+      register: email_result
+
+    - name: assert result changed
+      assert:
+        that:
+          - email_result.changed == False
+          - email_result.dkim_attributes.dkim_enabled == False
+
```

Comment:
Please test delete mode (including trying to re-delete the object)

Reply:
There is not delete mode as actions are just made on the existing identities. Hence also using the enable/disable terminology instead of present/absent (or add/delete). 

The deletion of the test objects is done in the always: section with the already existing `aws_ses_identity` module.

Reply:
What's the advantage of this being a separate module instead of extending aws_ses_identity?